WeldingState[] configurations
int32 current_index
bool auto_update